dateutil/parser/_parser.py

```diff
@@ -60,14 +60,8 @@ class _timelex(object):
     _split_decimal = re.compile("([.,])")
 
     def __init__(self, instream):
-        if six.PY2:
-            # In Python 2, we can't duck type properly because unicode has
-            # a 'decode' function, and we'd be double-decoding
-            if isinstance(instream, (bytes, bytearray)):
-                instream = instream.decode()
-        else:
-            if getattr(instream, 'decode', None) is not None:
-                instream = instream.decode()
+        if isinstance(instream, (bytes, bytearray)) or six.PY3 and isinstance(instream, memoryview):
```

Comment:
At the same time, I see that commit 8e786465437b51123ab7644b8ce94b51d947c5b8 says "Add support for bytesarray and other bytes-like", not sure if there is any other case we are missing here than `bytes`, `bytearray` and `unicode`.

Let's see if the test suite catches anything.